src/main/java/twitterapp/src/services/TwitterAppService.java

```diff
@@ -145,7 +155,32 @@ public TwitterAppService(Twitter twitter){
 
     }
 
+    public Optional<List<TwitterPost>> getUserTimeline() throws TwitterAppException{
+        try {
+            if(cacheUserTimeline.get(TIMELINE_KEY).isPresent() == false){
+                Paging page = new Paging(1,25);
+                Optional<List<TwitterPost>> resultListTwitterPost = Optional.ofNullable(twitter.getUserTimeline(page).stream()
+                        .map(s -> new TwitterPost(s.getText(),
+                                s.getUser().getName(),
+                                s.getUser().getScreenName(),
+                                s.getUser().getProfileImageURL(),
+                                s.getCreatedAt(),
+                                Objects.toString(s.getId())))
+                        .collect(toList()));
+
+                cacheUserTimeline.put(TIMELINE_KEY, resultListTwitterPost);
+            }
+
+            return cacheUserTimeline.get(1);
+
 
+        } catch (Exception e) {
+            log.error("There was a problem on the server side.", e);
```

Comment:
Your lab is simple in that in only talks to one other server (Twitter4J) but how helpful would this log message be if this called even one more API?